Sources/AblyAssetTrackingCore/ResultHandler.swift

```diff
@@ -1,10 +1,68 @@
-/**
- Completion handler for operations
- */
-public typealias ResultHandler<T: Any> = (Result<T, ErrorInformation>) -> Void
+import Foundation
 
-public extension Result where Success == Void {
-    static var success: Result {
+public typealias ResultHandler = (AATResult) -> Void
+
+@objc
+public protocol Resultable {
+    var success: Any? { get }
+    var failure: ErrorInformation? { get }
+}
+
+@objcMembers
+public class AATResult: NSObject, Resultable {
+    public let success: Any?
+    public let failure: ErrorInformation?
+    
+    public static var success: AATResult {
+        return .init(success: Void(), failure: nil)
+    }
+    
+    public static func success(_ value: Any?) -> AATResult {
+        return .init(success: value, failure: nil)
+    }
+    
+    public static func failure(_ value: ErrorInformation?) -> AATResult {
+        return .init(success: nil, failure: value)
+    }
+    
+    public init(success: Any? = nil, failure: ErrorInformation? = nil) {
+        self.success = success
+        self.failure = failure
+    }
+    
+    public override init() {
+        self.success = nil
+        self.failure = nil
+        
+        super.init()
+    }
+}
+
+public extension AATResult {
+    enum Unwrapped<T> {
```

Comment:
Could we not use `Swift.Result<T, ErrorInformation>` instead of this enum? And call the corresponding properties / functions `swiftResult` and `swiftResult(withSuccessType:)`, perhaps.

Reply:
2174f9efce35991624694e40798cb7e24fc44acd

Reply:
2174f9efce35991624694e40798cb7e24fc44acd